channels/offline-channel-js/src/InMemoryBatch.ts

```diff
@@ -0,0 +1,122 @@
+import dynamicProto from "@microsoft/dynamicproto-js";
+import {
+    IDiagnosticLogger, ITelemetryItem, _eInternalMessageId, _throwInternal, eLoggingSeverity, isNullOrUndefined
+} from "@microsoft/applicationinsights-core-js";
+import { IInMemoryBatch, IPostTransmissionTelemetryItem } from "./Interfaces/IInMemoryBatch";
+
+export class InMemoryBatch implements IInMemoryBatch {
+
+
+    constructor(logger: IDiagnosticLogger, endpoint: string, evts?: IPostTransmissionTelemetryItem[], evtsLimitInMem?: number) {
+        let _buffer: IPostTransmissionTelemetryItem[] = evts ? [].concat(evts) : [];
+        let _bufferFullMessageSent = false;
+
+        dynamicProto(InMemoryBatch, this, (_self) => {
+
+
+            _self.endpoint = () =>{
+                return endpoint;
+            }
+
+            _self["_getDbgPlgTargets"] = () => {
+                return [_bufferFullMessageSent];
+            };
+
+            
+            _self.addEvent = (payload: IPostTransmissionTelemetryItem | ITelemetryItem) => {
+                //TODO: handle space free up, handle drop here
+                if (!isNullOrUndefined(evtsLimitInMem) && _self.count() >= evtsLimitInMem) {
+                    // sent internal log only once
+                    if (!_bufferFullMessageSent) {
+                        _throwInternal(logger,
+                            eLoggingSeverity.WARNING,
+                            _eInternalMessageId.InMemoryStorageBufferFull,
+                            "Maximum offline in-memory buffer count reached: " + _self.count(),
+                            true);
+                        _bufferFullMessageSent = true;
+                    }
+
+                    return;
```

Comment:
We should probably return true/false so the caller knows whether an event was added (or not)...

So then perhaps the caller can do something (like flush / start a new buffer) when this returns false....